src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public void setKey(K key) {
+            this.key = key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public Node<K, V> getNextNode() {
+            return nextNode;
+        }
+
+        public void setNextNode(Node<K, V> nextNode) {
+            this.nextNode = nextNode;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            Node<K, V>[] oldBucketStock = values;
+            values = new Node[values.length * 2];
+            for (int i = 0; i < oldBucketStock.length; i++) {
+                while (oldBucketStock[i] != null) {
+                    K k = oldBucketStock[i].key;
+                    V v = oldBucketStock[i].value;
+                    int index = getIndexFromHash(k);
+                    if (values[index] == null) {
+                        values[index] = new Node<>(k, v);
+                    } else {
+                        Node<K, V> lastBucketNode = values[index];
+                        while (lastBucketNode.nextNode != null) {
+                            lastBucketNode = lastBucketNode.nextNode;
+                        }
+                        lastBucketNode.nextNode = new Node<>(k, v);
+                    }
+                    oldBucketStock[i] = oldBucketStock[i].nextNode;
+                }
+            }
+            this.threshold = (int) (values.length * LOAD_FACTOR);
+        }
+        int index = getIndexFromHash(key);
+        if (values[index] == null) {
+            values[index] = new Node<>(key, value);
+        } else {
+            Node<K, V> lastBucketNode = values[index];
+            if (lastBucketNode.key == null && key == null) {
+                lastBucketNode.value = value;
+                return;
+            } else if (key != null && key.equals(lastBucketNode.key)) {
+                lastBucketNode.value = value;
+                return;
+            }
+            while (lastBucketNode.nextNode != null) {
+                if (Objects.equals(lastBucketNode.key, key)) {
+                    lastBucketNode.value = value;
+                    return;
+                }
+                lastBucketNode = lastBucketNode.nextNode;
+            }
+            lastBucketNode.nextNode = new Node<>(key, value);
+        }
+        size = size + 1;
     }
 
+
+
     @Override
     public V getValue(K key) {
+        int index = getIndexFromHash(key);
+        Node<K, V> returnNode = values[index];
+        while (returnNode != null) {
+            /* if (key == returnNode.key) {
+                return returnNode.value;
+            } */
+            if (key == null && returnNode.key == null) {
+                return returnNode.value;
+            }
+            if (key != null && key.equals(returnNode.key)) {
```

Comment:
two if is bad because they do almost the same. it can be optimize
like:
if (Objects.equals(returnNode.key, key)) {
            return returnNode.value;
        }
        returnNode= returnNode.nextNode;